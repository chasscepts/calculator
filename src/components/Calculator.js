import React, { useState } from 'react';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import calculate from '../logic/calculate';
import AppHeader from './AppHeader';

const styles = {
  container: {
    width: '700px',
    margin: 'auto',
  },
  content: {
    display: 'flex',
    paddingBottom: '30px',
  },
};

export function Calculator() {
  const [total, setTotal] = useState(null);
  const [next, setNext] = useState(null);
  const [operation, setOperation] = useState(null);

  const handleClick = (buttonName) => {
    const state = calculate({ total, next, operation }, buttonName);
    setTotal(state.total);
    setNext(state.next);
    setOperation(state.operation);
  };

  return (
    <div style={styles.container}>
      <Display result={next || total || '0'} />
      <ButtonPanel clickHandler={handleClick} />
    </div>
  );
}

const p = "Now let's do some Math!";

export default function Page() {
  return (
    <>
      <AppHeader />
      <div style={styles.content} className="content">
        <p>{p}</p>
        <Calculator />
      </div>
    </>
  );
}
