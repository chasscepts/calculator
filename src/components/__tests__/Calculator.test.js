import React from 'react';
import renderer from 'react-test-renderer';
import ComponentWrapper from '../test_helpers/ComponentWrapper';
import Calculator from '../Calculator';

it('renders correctly', () => {
  const tree = renderer
    .create(<ComponentWrapper component={Calculator} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
