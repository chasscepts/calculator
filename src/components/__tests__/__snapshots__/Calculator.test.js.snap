// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-between",
        "padding": "15px 30px",
      }
    }
  >
    <h1>
      Math Magicians
    </h1>
    <nav>
      <ul
        style={
          Object {
            "display": "flex",
            "listStyle": "none",
          }
        }
      >
        <li
          style={
            Object {
              "padding": "0 10px",
            }
          }
        >
          <a
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li
          style={
            Object {
              "borderLeft": "1px solid #ddd",
              "borderRight": "1px solid #ddd",
              "padding": "0 10px",
            }
          }
        >
          <a
            href="/calculator"
            onClick={[Function]}
          >
            Calculator
          </a>
        </li>
        <li
          style={
            Object {
              "padding": "0 10px",
            }
          }
        >
          <a
            href="/quote-of-the-day"
            onClick={[Function]}
          >
            Quote
          </a>
        </li>
      </ul>
    </nav>
  </div>,
  <div
    className="content"
    style={
      Object {
        "display": "flex",
        "paddingBottom": "30px",
      }
    }
  >
    <p>
      Now let's do some Math!
    </p>
    <div
      style={
        Object {
          "margin": "auto",
          "width": "700px",
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "gray",
            "color": "#fff",
            "display": "flex",
            "flexDirection": "column",
            "fontSize": "48px",
            "fontWeight": "bold",
            "height": "100px",
            "justifyContent": "center",
            "padding": "0 10px",
            "textAlign": "right",
            "width": "100%",
          }
        }
      >
        <span>
          0
        </span>
      </div>
      <div>
        <div
          style={
            Object {
              "display": "flex",
              "height": "100px",
              "width": "100%",
            }
          }
        >
          <button
            onClick={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ddd",
                "border": "2px solid gray",
                "color": "#000",
                "display": "flex",
                "flex": "0 0 25%",
                "fontSize": "24px",
                "fontWeight": "bold",
                "justifyContent": "center",
              }
            }
            type="button"
          >
            <span>
              AC
            </span>
          </button>
          <button
            onClick={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ddd",
                "border": "2px solid gray",
                "color": "#000",
                "display": "flex",
                "flex": "0 0 25%",
                "fontSize": "24px",
                "fontWeight": "bold",
                "justifyContent": "center",
              }
            }
            type="button"
          >
            <span>
              +/-
            </span>
          </button>
          <button
            onClick={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ddd",
                "border": "2px solid gray",
                "color": "#000",
                "display": "flex",
                "flex": "0 0 25%",
                "fontSize": "24px",
                "fontWeight": "bold",
                "justifyContent": "center",
              }
            }
            type="button"
          >
            <span>
              %
            </span>
          </button>
          <button
            onClick={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "orange",
                "border": "2px solid gray",
                "color": "#000",
                "display": "flex",
                "flex": "0 0 25%",
                "fontSize": "24px",
                "fontWeight": "bold",
                "justifyContent": "center",
              }
            }
            type="button"
          >
            <span>
              รท
            </span>
          </button>
        </div>
        <div
          style={
            Object {
              "display": "flex",
              "height": "100px",
              "width": "100%",
            }
          }
        >
          <button
            onClick={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ddd",
                "border": "2px solid gray",
                "color": "#000",
                "display": "flex",
                "flex": "0 0 25%",
                "fontSize": "24px",
                "fontWeight": "bold",
                "justifyContent": "center",
              }
            }
            type="button"
          >
            <span>
              7
            </span>
          </button>
          <button
            onClick={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ddd",
                "border": "2px solid gray",
                "color": "#000",
                "display": "flex",
                "flex": "0 0 25%",
                "fontSize": "24px",
                "fontWeight": "bold",
                "justifyContent": "center",
              }
            }
            type="button"
          >
            <span>
              8
            </span>
          </button>
          <button
            onClick={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ddd",
                "border": "2px solid gray",
                "color": "#000",
                "display": "flex",
                "flex": "0 0 25%",
                "fontSize": "24px",
                "fontWeight": "bold",
                "justifyContent": "center",
              }
            }
            type="button"
          >
            <span>
              9
            </span>
          </button>
          <button
            onClick={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "orange",
                "border": "2px solid gray",
                "color": "#000",
                "display": "flex",
                "flex": "0 0 25%",
                "fontSize": "24px",
                "fontWeight": "bold",
                "justifyContent": "center",
              }
            }
            type="button"
          >
            <span>
              X
            </span>
          </button>
        </div>
        <div
          style={
            Object {
              "display": "flex",
              "height": "100px",
              "width": "100%",
            }
          }
        >
          <button
            onClick={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ddd",
                "border": "2px solid gray",
                "color": "#000",
                "display": "flex",
                "flex": "0 0 25%",
                "fontSize": "24px",
                "fontWeight": "bold",
                "justifyContent": "center",
              }
            }
            type="button"
          >
            <span>
              4
            </span>
          </button>
          <button
            onClick={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ddd",
                "border": "2px solid gray",
                "color": "#000",
                "display": "flex",
                "flex": "0 0 25%",
                "fontSize": "24px",
                "fontWeight": "bold",
                "justifyContent": "center",
              }
            }
            type="button"
          >
            <span>
              5
            </span>
          </button>
          <button
            onClick={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ddd",
                "border": "2px solid gray",
                "color": "#000",
                "display": "flex",
                "flex": "0 0 25%",
                "fontSize": "24px",
                "fontWeight": "bold",
                "justifyContent": "center",
              }
            }
            type="button"
          >
            <span>
              6
            </span>
          </button>
          <button
            onClick={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "orange",
                "border": "2px solid gray",
                "color": "#000",
                "display": "flex",
                "flex": "0 0 25%",
                "fontSize": "24px",
                "fontWeight": "bold",
                "justifyContent": "center",
              }
            }
            type="button"
          >
            <span>
              +
            </span>
          </button>
        </div>
        <div
          style={
            Object {
              "display": "flex",
              "height": "100px",
              "width": "100%",
            }
          }
        >
          <button
            onClick={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ddd",
                "border": "2px solid gray",
                "color": "#000",
                "display": "flex",
                "flex": "0 0 25%",
                "fontSize": "24px",
                "fontWeight": "bold",
                "justifyContent": "center",
              }
            }
            type="button"
          >
            <span>
              1
            </span>
          </button>
          <button
            onClick={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ddd",
                "border": "2px solid gray",
                "color": "#000",
                "display": "flex",
                "flex": "0 0 25%",
                "fontSize": "24px",
                "fontWeight": "bold",
                "justifyContent": "center",
              }
            }
            type="button"
          >
            <span>
              2
            </span>
          </button>
          <button
            onClick={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ddd",
                "border": "2px solid gray",
                "color": "#000",
                "display": "flex",
                "flex": "0 0 25%",
                "fontSize": "24px",
                "fontWeight": "bold",
                "justifyContent": "center",
              }
            }
            type="button"
          >
            <span>
              3
            </span>
          </button>
          <button
            onClick={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "orange",
                "border": "2px solid gray",
                "color": "#000",
                "display": "flex",
                "flex": "0 0 25%",
                "fontSize": "24px",
                "fontWeight": "bold",
                "justifyContent": "center",
              }
            }
            type="button"
          >
            <span>
              -
            </span>
          </button>
        </div>
        <div
          style={
            Object {
              "display": "flex",
              "height": "100px",
              "width": "100%",
            }
          }
        >
          <button
            onClick={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ddd",
                "border": "2px solid gray",
                "color": "#000",
                "display": "flex",
                "flex": "0 0 50%",
                "fontSize": "24px",
                "fontWeight": "bold",
                "justifyContent": "center",
              }
            }
            type="button"
          >
            <span>
              0
            </span>
          </button>
          <button
            onClick={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ddd",
                "border": "2px solid gray",
                "color": "#000",
                "display": "flex",
                "flex": "0 0 25%",
                "fontSize": "24px",
                "fontWeight": "bold",
                "justifyContent": "center",
              }
            }
            type="button"
          >
            <span>
              .
            </span>
          </button>
          <button
            onClick={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "orange",
                "border": "2px solid gray",
                "color": "#000",
                "display": "flex",
                "flex": "0 0 25%",
                "fontSize": "24px",
                "fontWeight": "bold",
                "justifyContent": "center",
              }
            }
            type="button"
          >
            <span>
              =
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>,
]
`;
